bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/HBaseImportIO.java

```diff
@@ -0,0 +1,159 @@
+ * hbase-server jars.
+ *
+ * The following code snippets sets up and runs an import pipeline:
+ *
+ * <pre>
+ * {@code
+ *   HBaseImportOptions options =
+ *       PipelineOptionsFactory.fromArgs(args).withValidation().as(HBaseImportOptions.class);
+ *   Pipeline p = HBaseImportIO.createImportPipeline(options);
+ *   p
+ *       .apply("ReadSequenceFile", Read.from(HBaseImportIO.read(options)))
+ *       .apply("ConvertResultToMutations", HBaseImportIO.processRow(options))
+ *       .apply("WriteToTable", HBaseImportIO.writeToTable(options));
+ *   p.run();
+ * }
+ * </pre>
+ */
+public class HBaseImportIO {
+  private static final Logger logger = LoggerFactory.getLogger(HBaseImportIO.class);
+
+  // Needed for HBase 0.94 format. Copied from ResultSerialization.IMPORT_FORMAT_VER.
+  private static final String IMPORT_FORMAT_VER = "hbase.import.version";
+  private static final String VERSION_094_STRING = "0.94";
+
+  /**
+   * Returns a {@link Pipeline} that may write to a Cloud Bigtable.
+   */
+  public static Pipeline createImportPipeline(HBaseImportOptions options) {
+    return CloudBigtableIO.initializeForWrite(Pipeline.create(options));
+  }
+
+  /**
+   * Returns a {@link BoundedSource} from an HBase Sequence File for an import pipeline.
+   */
+  public static BoundedSource<KV<ImmutableBytesWritable, Result>> read(HBaseImportOptions options) {
+    return HadoopFileSource.from(
+        options.getFilePattern(),
+        SequenceFileInputFormat.class,
+        ImmutableBytesWritable.class,
+        Result.class,
+        KvCoder.of(new WritableCoder<>(ImmutableBytesWritable.class), new HBaseResultCoder()),
+        createSerializationProperties(options.isHBase094DataFormat()));
+  }
+
+  /**
+   * <p>Creates a {@link PTransform} that can write either a bounded or unbounded
+   * {@link PCollection} of {@link Mutation}s ({@code Put}s and {@code Delete}s) to a table
+   * specified via a {@link HBaseImportOptions}.
+   */
+  public static PTransform<PCollection<Mutation>, PDone> writeToTable(HBaseImportOptions options) {
+    return CloudBigtableIO.writeToTable(CloudBigtableTableConfiguration.fromCBTOptions(options));
+  }
+
+  /**
+   * <p>
+   * Returns a {@link PTransform} that converts a {@link Result} into
+   * {@link Mutation}s.
+   */
+  public static PTransform<
+      PCollection<? extends KV<ImmutableBytesWritable, Result>>,
+      PCollection<Mutation>> processRow(HBaseImportOptions options) {
+    return ParDo.of(new HBaseResultToMutationFn(options));
+  }
+
+
+  private static ImmutableMap<String, String> createSerializationProperties(
+      boolean isHBase094DataFormat) {
+    if (isHBase094DataFormat) {
+      return ImmutableMap.of(IMPORT_FORMAT_VER, VERSION_094_STRING);
+    } else {
+      return ImmutableMap.of();
+    }
+  }
+
+  /**
+   * A {@link DoFn} function that converts a {@link Result} in the pipeline input to a
+   * {@code Mutation} for output.
+   */
+  public static class HBaseResultToMutationFn
```

Comment:
Try to avoid having more than one way of doing things. If we want user to directly use this class, then remove transformToMutations(HBaseImportOptions options).